packages/@lwc/ssr-compiler/runtime.js

```diff
@@ -0,0 +1,137 @@
+const MULTI_SPACE = /\s+/g;
+
+class ClassList {
+    constructor(el) {
+        this.el = el;
+    }
+
+    add(...newClassNames) {
+        const className = this.el.className;
+        const set = new Set(className.split(MULTI_SPACE).filter(Boolean));
+        for (const newClassName of newClassNames) {
+            set.add(newClassName);
+        }
+        this.el.className = Array.from(set).join(' ');
+    }
+
+    contains(className) {
+        const currentClassNameStr = this.el.className;
+        return new RegExp(`\\b${className}\\b`).test(currentClassNameStr);
```

Comment:
```suggestion
        return currentClassNameStr.split(MULTI_SPACE).includes(className);
```

Not great for perf, but saves us from regex DOS and from having to escape the strings for the regex. Can be optimized later.